src/util/evalUnits.ts

```diff
@@ -53,19 +63,51 @@ export enum EvalMessages {
  * the first is the actual calculated number (or null if none) while
  * the second is the message that should appear in case of error
  */
-export function evalUnits(input: string): [number | null, string] {
-  if (!floats.test(input) && !ints.test(input) && !alphaInts.test(input) && !alphaFloats.test(input)) {
+export function evalUnits(input: string, chainDecimals: number): [BN | null, string] {
+  //sanitize input to remove + char if exists
+  input = input && input.replace('+', '');
+  if (
+    !new RegExp(floats).test(input) &&
+    !new RegExp(ints).test(input) &&
+    !new RegExp(alphaInts).test(input) &&
+    !new RegExp(alphaFloats).test(input)
```

Comment:
Instead of creating a new `RegExp` every time, you could just make the consts above the function a `RegExp` right away.